src/krux/wallet.py

```diff
@@ -168,14 +190,51 @@ def load(self, wallet_data, qr_format, allow_assumption=None):
                 "n": n,
                 "cosigners": cosigners,
             }
-        elif self.descriptor.miniscript is not None:
+        elif self.descriptor.miniscript is not None or self.descriptor.taptree:
+            if self.descriptor.taptree:
+                if not descriptor.keys[0].origin:
+                    import hashlib
+                    from embit.ec import PublicKey
```

Comment:
could also import NUMS_PUBKEY directly from embit.ec here, and use it below instead of redefining BIP_341_NUMS_EXAMPLE at line 45 above.

Note: embit.ec.NUMS_PUBKEY has comment pointing to bip341... but your added comments above might be kept somewhere here.

Reply:
Great finding! Let's use it!